client/components/data/query-users/index.jsx

```diff
@@ -37,12 +39,18 @@ class QueryUsers extends Component {
 	}
 
 	request() {
-		this.props.requestUsers( this.props.siteId, this.props.userIds );
+		const { siteId, userIds, currentUserId } = this.props;
+		if ( userIds.length ) {
```

Comment:
Only request users if a list of user Ids is provided. Otherwise, all users would be requested incl. the current user.

Reply:
I'd leave context like these in inline comments bc it quickly brought me up to speed with your intentions when I read the code.